loader/loader.go

```diff
@@ -120,16 +120,16 @@ func (l *loader) GetMatchingReleasePlanAdmission(ctx context.Context, cli client
 		return releasePlanAdmission, toolkit.GetObject(designatedReleasePlanAdmissionName, releasePlan.Spec.Target, cli, ctx, releasePlanAdmission)
 	}
 
-	if releasePlan.Spec.Target == "" {
-		return nil, fmt.Errorf("releasePlan has no target, so no ReleasePlanAdmissions can be found")
-	}
-
 	releasePlanAdmissions := &v1alpha1.ReleasePlanAdmissionList{}
-	err := cli.List(ctx, releasePlanAdmissions,
-		client.InNamespace(releasePlan.Spec.Target),
-		client.MatchingFields{"spec.origin": releasePlan.Namespace})
-	if err != nil {
-		return nil, err
+	if releasePlan.Spec.Target != "" {
```

Comment:
I think it's better to exit earlier just by checking for `Target == ""`. That way you avoid the else block altogether.

Reply:
Ah, I probably misunderstood the ask in the acceptance criteria.

@davidmogar  do you mean same as we already have  [Line 123](https://github.com/konflux-ci/release-service/pull/583/files#diff-8e9af1e483632eac17610237caaa5f879ea7bfb3e1e1ea228e45a17c770b9432L123)?
Just need to add checks to other calls same way as done already for above.

Reply:
I'm confused.This function seemingly already exits early if target is `""`
```
	if releasePlan.Spec.Target == "" {
		return nil, fmt.Errorf("releasePlan has no target, so no ReleasePlanAdmissions can be found")
	}

	releasePlanAdmissions := &v1alpha1.ReleasePlanAdmissionList{}
	err := cli.List(ctx, releasePlanAdmissions,
		client.InNamespace(releasePlan.Spec.Target),
		client.MatchingFields{"spec.origin": releasePlan.Namespace})
	if err != nil {
		return nil, err
	}
```
not sure there is anything to fix in this one?